### PR TITLE
add &clash minigame

### PR DESCRIPTION
**Pull request by Felix-44**
basically Clash of Clans but more simple

### CODE REVIEW

**Comment by Felix-44** [comment]
pretty sure I finished this? idk actually but it's ready for review I think

File: src/Minigames/clash.js [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+                message.channel.send("Now that position you just gave me...it doesn't make sense")
+                return
+            }
+            if (!ClashMatrix.checkBounds(x, y)) {
+                message.channel.send("Now that position you just told me...it's out of bounds")
+                return
+            }
+			if (!ClashMatrix.compare(x, y, "N")) {
+				return void message.channel.send("Dude you're gonna build on top of something else, move it first")
+			}
```
Comment (LJMSU) [inline src/Minigames/clash.js:224]: surreal indentation

File: src/Minigames/clash.js [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+                ClashMatrix.set(x, y, building)
+                if (building == "M") EconomySystem.clashtime = Date.now()
+            }
+            break
+        }
+        case "move": {
+            const xStart = parseInt(args[1])
+            const yStart = parseInt(args[2])
+            const xEnd = parseInt(args[3])
+            const yEnd = parseInt(args[4])
```
Comment (LJMSU) [inline src/Minigames/clash.js:236]: `const [xStart, yStart, xEnd, yEnd] = args.slice(1, 5).map((arg) => parseInt(arg))` *might* be a shorter way to write this.

**Review by LJMSU — APPROVED** [review]

File: src/Minigames/clash.js [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+        )
+        return MatrixEmbed
+    }
+
+    getTotal(building) {
+        let amount = 0
+        for (const Cell of this) {
+            if (Cell.value == building) amount++
+        }
+		return amount
```
[the remark this answers is not in the document]
Comment (Felix-44) [reply src/Minigames/clash.js:75]: what the
why does every text viewer thing it's 2 spaced????
if you put it to 4 tabs it looks alright

[the remark this answers is not in the document]
Comment (Felix-44) [reply src/Minigames/clash.js:75]: okay fixed

File: src/Minigames/clash.js (hunk 2 above)
Comment (Felix-44) [reply src/Minigames/clash.js]: aight I'll use that instead

**Pull request merged**